base/bucket_gocb.go

```
@@ -917,13 +923,13 @@ func (bucket CouchbaseBucketGoCB) GetWithXattr(k string, xattrKey string, rv int
 			// Successfully retrieved doc and (optionally) xattr.  Copy the contents into rv, xv and return.
 			contentErr := res.Content("", rv)
 			if contentErr != nil {
-				LogTo("gocb", "Unable to retrieve document content for key=%s, xattrKey=%s: %v", k, xattrKey, contentErr)
-				return false, contentErr, nil
+				LogTo("CRUD", "Unable to retrieve document content for key=%s, xattrKey=%s: %v", k, xattrKey, contentErr)
+				return false, contentErr, 0
```

This seems like a discrepancy to return a non-nil `contentErr` in conjuntion with a `0`, since the caller of the RetryLoop is doing this:

```
	err, result := RetryLoop(description, worker, bucket.spec.RetrySleeper())

	if result == nil {
		return 0, err
	}
```

And so it looks to me like it will be ignoring errors

---

RetryLoop is already returning 'err, nil' if there's a non-recoverable error, so I don't believe any errors will be ignored.  I think it makes sense for the worker to return the appropriate type for their operation, even if the value needs to be defined as interface{} for reuse across multiple operation types.

---

Ok that's surprising .. looking at the usage, I would have expected the RetryLoop to return whatever was returned from the `worker` callback.

Ok, now at least the code makes a little more sense.

I can see how it would work, but wouldn't this be clearer? 

```
if err != nil {
    return 0, err
}
```